python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py

```diff
@@ -0,0 +1,377 @@
+# Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the License is located at
+#
+#        http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for
+# the specific language governing permissions and limitations under the License.
+
+"""
+Description:
+  Checks that Amazon API Gateway APIs are of type as specified in the rule parameter 'endpointConfiguration Type'. The rule returns NON_COMPLIANT if the RestApi does not match the endpoint type configured in the rule parameter.
+
+Trigger:
+  Configuration Change on AWS::ApiGateway::RestApi
+
+Reports on:
+  AWS::ApiGateway::RestApi
+
+Rule Parameters:
+  endpointConfigurationType
+  (Required) Comma-separated list of allowed endpoint types.
+
+Scenarios:
+  Scenario: 1
+    Given: The rule parameter is not provided
+     Then: Return ERROR
+  Scenario: 2
+    Given: The rule parameter is invalid
+     Then: Return ERROR
+  Scenario: 3
+    Given: The rule parameter is provided and valid
+      And: The endpointConfiguration type in the RestApi configuration item does not match the endpoint type in rule parameter.
+     Then: Return NON_COMPLIANT
+  Scenario: 4
+    Given: The rule parameter is provided and valid
+      And: The endpointConfiguration type in the RestApi configuration item matches the endpoint type in rule parameter.
+     Then: Return COMPLIANT
+"""
+
+import json
+import sys
+import datetime
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::ApiGateway::RestApi'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if configuration_item['configuration']['endpointConfiguration']['types'][0] in valid_rule_parameters:
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+    return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation="This Amazon API Gateway endpoint type does not match as specified in rule parameter(endpointConfigurationType): " + str(valid_rule_parameters) + ".")
+
+def evaluate_parameters(rule_parameters):
+    if 'endpointConfigurationType' not in rule_parameters:
+        raise ValueError("endpointConfigurationType is a mandatory parameter.")
+    rule_parameters_list = rule_parameters['endpointConfigurationType'].split(',')

```

Comment:
please manage if there is an extra space. You can use strip() somewhere in your code